### DISCUSSION TITLE
<ctrl-r> behaves differently in "files" and "live_grep"

### DISCUSSION THREAD

**Discussion by i11010520** [ibhagwan/fzf-lua#1412] [Q&A]
I posted the [question in LazyVim](https://github.com/LazyVim/LazyVim/discussions/4320#discussion-7067599).
Some good man points out `<ctrl-r>` seems to conflict with `resume = true`.
And I validate that's true.

**Comment by ibhagwan** [comment]
I'n not sure what you mean, there's no `ctrl-r` mapping for anything in fzf-lua anywhere so I'm not sure why it would conflict with `resume=true`, it's either something in your setup (or LazyVim's defaults) or you need to provide more details with a proper example so I can help you.

**Reply by ibhagwan** [reply]
I read your linked thread, I'm not sure what `ctrl-r` supposed to do in LazyVim but by you adding `resume=true` under both `files` and `grep` you're asking fzf-lua to restore all call options with each call, that includes the last query, cwd, winopts, etc (all caller specified options), in this case if you're trying to change `cwd`  I can see why the resume call options would overwrite that.

What are you trying to do? why have you added `resume=true`  under `setup.{files|grep}`? Why don't you remove `resume=true` from setup and add it to the keybmap instead (i.e. `:FzfLua files resume=true`)? that way only the initial interface will be resumed but the actions `(e.g. `ctrl-r`) would still work.

**Comment by i11010520** [comment]
@ibhagwan Thanks for your quick response!

I wanna custom 'fzf-lua' in 'LazyVim' as follows:
For both 'files' mode and 'live_grep' mode, 
1) pressing `<ctrl-r>` can switch between `cwd` and `root` path, 
2) resuming with previous search input when re-open.

This is my custom script under LazyVim's config flavor 
(Sorry I cannot dive into LazyVim's transpilation now):
```
return {
  "ibhagwan/fzf-lua",
  opts = function(_, opts)
    return vim.tbl_deep_extend("force", opts, {
      fzf_opts = {
        ["--layout"] = "default",
      },

      files = {
        resume = true,
        cwd_prompt = true,
      },
      grep = {
        rg_glob = true,
        glob_flag = "--iglob",
        glob_separator = "%s%-%-",
        resume = true,
        cwd_prompt = true,
      },
    })
  end,
}
```
With above script, `<ctrl-r>` switching works just at "files" mode, not at "live_grep" mode.
If I set `grep.resume = false`, `<ctrl-r>` works at both modes. (However there's new doubt appeared that `grep.resume` value affect "files" mode's resuming state, which can be ignored temporarily.)

**Reply by ibhagwan** [reply]
I don’t think this is the right way to achieve that, I need to look at the code by resume will restore call options and therefore may override your cwd, not sure why the difference between files and live grep but I recommend to remove `resume=true` from your setup options and add it to your mapping instead, e.g. `:FzfLua live_grep resume=true`.

**Reply by i11010520** [reply]
Appreciate!

I try it later for resume setting.

**Reply by ibhagwan** [reply]
@i11010520, not sure how `ctrl-r` is configured in LazyVim, just tried this simple `cwd` toggle with latest commit (for the dynamic header fix) and it works as expected:
```lua
require("fzf-lua").setup({
  actions = {
    files = {
      ["ctrl-r"] = {
        header = function(o)
          return string.format("switch to '%s'", o.cwd ~= "~" and "~" or "~/.config/nvim")
        end,
        fn = function(_, o)
          require("fzf-lua").resume({
            cwd = o.cwd ~= vim.fn.expand("$HOME") and "~" or "~/.config/nvim"
          })
        end,
      },
    }
  },
  files = { resume = true },
  grep = { resume = true },
})
```

Files before `ctrl-r`:
![image](https://github.com/user-attachments/assets/579a932c-ddc1-4dcf-a84b-7f9dfadf176b)

Files after `ctrl-r`:
![image](https://github.com/user-attachments/assets/37960959-8bb1-4e26-b426-6aee604b5d99)

Live grep before `ctrl-r`:
![image](https://github.com/user-attachments/assets/f286094c-1fb4-456b-80d8-ff4fce887903)

Live grep after `ctrl-r`:
![image](https://github.com/user-attachments/assets/49c98139-294e-43fa-9326-7ed8d1e99266)
